test/controller/company.number.controller.spec.unit.ts

```diff
@@ -0,0 +1,100 @@
+import request from "supertest";
+import app from "../../src/app";
+import authenticationMiddleware from "../../src/middleware/authentication.middleware";
+import sessionMiddleware from "../../src/middleware/session.middleware";
+import ObjectionCompanyProfile from "../../src/model/objection.company.profile";
+import { COMPANY_NUMBER, OBJECTIONS_COMPANY_NUMBER, OBJECTIONS_CONFIRM_COMPANY } from "../../src/model/page.urls";
+import { getCompanyProfile } from "../../src/services/company.profile.service";
+import { COOKIE_NAME } from "../../src/utils/properties";
+
+import { getValidAccessToken } from "../../src/services/objections.session.service";
+
+const ACCESS_TOKEN = "KGGGUYUYJHHVK1234";
+
+const mockAuthenticationMiddleware = authenticationMiddleware as jest.Mock;
+mockAuthenticationMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
+
+const mockSessionMiddleware = sessionMiddleware as jest.Mock;
+mockSessionMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => {
+    req.session = {
+        data: {},
+    } as Session;
+    return next();
+});
+
+describe("company number lookup tests", () => {
+
+    const mockCompanyProfile = getCompanyProfile as jest.Mock;
+
+    beforeEach(() => {
+        mockCompanyProfile.mockReset();
+    });
+
+    // TODO - Implement generic error page when company search failure handling is implemented
+    //        based on story requirements
+
+    // it("should return the error page when company search fails", async () => {
+    //     mockCompanyProfile.mockImplementation(() => {
+    //         throw {
+    //             message: "Unexpected error",
+    //             status: 500,
+    //         };
+    //     });
+
+    //     const response = await request(app)
+    //         .post(OBJECTIONS_COMPANY_NUMBER)
+    //         .set("Accept", "application/json")
+    //         .set("Referer", "/")
+    //         .set("Cookie", [`${COOKIE_NAME}=123`])
+    //         .send({companyNumber: "00012345"});
+
+    //     expect(response.status).toEqual(500);
+    //     expect(response).not.toBeUndefined();
+    // });
+
+    it("should redirect to the check company details screen when company is found", async () => {
+
+        const mockValidAccessToken = getValidAccessToken as jest.Mock;
+
+        beforeEach(() => {
+            mockValidAccessToken.mockReset();
+        });
+
+        mockValidAccessToken.mockReset();
+        mockValidAccessToken.mockImplementation(() => ACCESS_TOKEN );
+
+        mockCompanyProfile.mockResolvedValueOnce(dummyCompanyProfile);
+
+        const response = await request(app)
+            .post(OBJECTIONS_COMPANY_NUMBER)
+            .set("Accept", "application/json")
+            .set("Referer", "/")
+            .set("Cookie", [`${COOKIE_NAME}=123`])
+            .send({companyNumber: COMPANY_NUMBER});
+
+        expect(response.header.location).toEqual(OBJECTIONS_CONFIRM_COMPANY);
```

Comment:
I would also add a couple of expects to check company info is on the page correctly like you've done for confirm company

Reply:
This is why I didn't add any expects because we are covering this with the other test fopr the company confirmation screen.

Reply:
sorry I got it wrong, the company details aren't shown on that screen, I was confused with the confirm screen. Nothing to do here.